Source/inv.cpp

```diff
@@ -450,6 +450,99 @@ void DrawInvBelt()
 	}
 }
 
+BOOL IsEmpty(const ItemStruct &item)
+{
+	return item._itype == ITYPE_NONE;
+}
+
+BOOL CanEquip(const ItemStruct &item)
+{
+	return item._iLoc != ILOC_INVALID && item._iLoc != ILOC_NONE && item._iLoc != ILOC_UNEQUIPABLE && item._iStatFlag;
+}
+
+BOOL CanWield(int playerNumber, const ItemStruct &item)
+{
+	if (!CanEquip(item) || (item._iLoc != ILOC_ONEHAND && item._iLoc != ILOC_TWOHAND))
+		return FALSE;
+
+	PlayerStruct &player = plr[playerNumber];
+	ItemStruct &leftHandItem = player.InvBody[INVLOC_HAND_LEFT];
+	ItemStruct &rightHandItem = player.InvBody[INVLOC_HAND_RIGHT];
+
+	if (IsEmpty(leftHandItem) && IsEmpty(rightHandItem)) {
+		return TRUE;
+	}
+
+	if (!IsEmpty(leftHandItem) && !IsEmpty(rightHandItem)) {
+		return FALSE;
+	}
+
+	ItemStruct &occupiedHand = !IsEmpty(leftHandItem) ? leftHandItem : rightHandItem;
+
+	return item._iLoc != ILOC_TWOHAND && occupiedHand._iLoc != ILOC_TWOHAND && item._iClass != occupiedHand._iClass;
+}
+
+BOOL CanEquip(int playerNumber, const ItemStruct &item, int bodyLocation)
+{
+	PlayerStruct &player = plr[playerNumber];
+	if (!CanEquip(item) || player._pmode > PM_WALK3 || !IsEmpty(player.InvBody[bodyLocation]))
+		return FALSE;
+
+	switch (bodyLocation) {
+	case INVLOC_AMULET:
+		return item._iLoc == ILOC_AMULET;
+
+	case INVLOC_CHEST:
+		return item._iLoc == ILOC_ARMOR;
+
+	case INVLOC_HAND_LEFT: 
+	case INVLOC_HAND_RIGHT:
+		return CanWield(playerNumber, item);
+
+	case INVLOC_HEAD:
+		return item._iLoc == ILOC_HELM;
+
+	case INVLOC_RING_LEFT:
+	case INVLOC_RING_RIGHT:
+		return item._iLoc == ILOC_RING;
+
+	default:
+		return FALSE;
+	}
+}
+
+BOOL AutoEquip(int playerNumber, const ItemStruct &item, int bodyLocation)
+{
+	if (!CanEquip(playerNumber, item, bodyLocation)) {
+		return FALSE;
+	}
+
+	plr[playerNumber].InvBody[bodyLocation] = item;
+
+	NetSendCmdChItem(FALSE, bodyLocation);
+	CalcPlrInv(playerNumber, TRUE);
+	if (playerNumber == myplr) {
+		PlaySFX(ItemInvSnds[ItemCAnimTbl[item._iCurs]]);

```

Comment:
If you can remove the added sfx I think I'm ok merging it for 1.2.0

Reply:
Can you elaborate a bit on why? I felt like adding the sounds resulted in a really nice experience here and is consistent with the sound it makes when you manually equip the item.

I feel like in this case it improves on the game (without being invasive) while resulting in simpler code: every call to equip will play the sound unconditionally (I intend to use these methods for other things later).

Let me know your thoughts.

Reply:
Because I want to be as conservative as possible when changing behaviour. Atm. this PR makes one consisten, but also introduces a new one.

Adding audio feedback is nice, yes. But it's not something the game does. Adding this open the gate for also having sounds when you level up, items break, stafs run out of charges etc.

Reply:
Commit for playing audio removed from the branch.